quartz/util/encrypt.ts

```diff
@@ -0,0 +1,33 @@
+import { base64 } from "rfc4648"
+import { getRandomValues, subtle } from "crypto"
+
+export async function getEncryptedPayload(
+  content: string,
+  password: string,
+  iterations: number = 2e6,
+) {
+  const encoder = new TextEncoder()
+  const salt = getRandomValues(new Uint8Array(32))
```

Comment:
Does this random salt mean that every time you rebuild, you get a different result? How does the decryption account for that?

Reply:
Yes, salt will be changed when you rebuild to enhance security. I used PBKDF2 for key derivation and AES-GCM for the encryption algorithm.

With this encryption method, salt, iv and cipher will be merged into an array of 8-bit unsigned integers, and encode with base64. When decrypting, salt, iv and cipher will be decoded, extracted with same sequence.

Reply:
You can check it from this [reference link](https://gist.github.com/mrdarrengriffin/7e0046031e46adbe9a1b40566534c300)

Reply:
Btw, I'm considering to use Argon2 instead of PBKDF2 because it's better

Reply:
I just noticed you prepend the salt and iv in front of the hashed page. I was wondering how you extract the salt if it is random, but that explains it.